docs/design/document-type.md

```diff
@@ -14,346 +14,94 @@ such as `struct`, `union`, etc. which are handled by the [Kotlin Smithy SDK](kot
 
 # Design
 
-The `document` type is an untyped JSON-like value that can take on the following types: null, boolean, string, byte,
-short, integer, long, float, double, an array of these types, or a map of these types where the key is string.
+The `document` type is an untyped JSON-like value that can take on the following types:
+* null
+* boolean
+* string
+* number (byte, short, integer, long, float, double)
```

Comment:
We should clarify arbitrary precision number is called out in spec and in scope.